tests/test_aliases.py

```diff
@@ -335,3 +355,45 @@ def alias_generator(x):
         'd_config_parent',
         'e_generator_child',
     ]
+
+
+@pytest.mark.parametrize(
+    [
+        'use_construct',
+        'allow_population_by_field_name_config',
+        'arg_name',
+        'expectation',
+    ],
+    [
+        [False, True, 'bar', does_not_raise()],
+        [False, True, 'bar_', does_not_raise()],
+        [False, False, 'bar', does_not_raise()],
+        [False, False, 'bar_', pytest.raises(ValueError)],
+        [True, True, 'bar', does_not_raise()],
+        [True, True, 'bar_', does_not_raise()],
+        [True, False, 'bar', does_not_raise()],
+        [True, False, 'bar_', does_not_raise()],
```

Comment:
If construct were to behave consistently with respect to `allow_population_by_field_name_config`, this would raise a `ValueError`, which is how I had the behavior in my initial commit, but that would be a breaking change, so I decided against it.